tests/unit/mock_hostmod.c

```diff
@@ -233,9 +269,16 @@ osd_result osd_hostmod_reg_write(struct osd_hostmod_ctx *ctx,
         ck_assert_int_eq(exp->flags, flags);
     }
 
-    assert(reg_size_bit == 16); // XXX: Extend for other register sizes
-    uint64_t written_reg_val = *(uint16_t*)reg_val;
-    ck_assert_uint_eq(written_reg_val, exp->reg_val);
+    assert(reg_size_bit%16 == 0 && reg_size_bit<=32); // XXX: Extend for other register sizes (64 and 128 bits)
```

Comment:
That's clever, but harder to read than simply checking for 16 or 32. Readability of code is key!

Reply:
Should I change it to 
```
assert(reg_size_bit%16 == 0 && reg_size_bit<=128);
```
or check for each size individually?

Reply:
Check for each size individually, it's the most explicit solution and hence most readable.